crates/dyn-abi/src/ty.rs

```diff
@@ -937,4 +937,41 @@ mod tests {
             0000000000000000000000000000000000000000000000000000000000001337
         "),
     }
+    #[test]
+    fn test_hash_implementation() {
+        // Create instances for DynSolType
+        let int_42 = DynSolType::Int(42);
+        let int_43 = DynSolType::Int(43);
+        let bool_type = DynSolType::Bool;
+
+        // Create instances for StructProp
+        let prop_a = StructProp { name: "prop".to_string(), ty: int_42.clone() };
```

Comment:
you'll need to import alloc::string::ToString

Reply:
ok, i'll do it later today and send another pr